Add support for Flattening (union) PCollections and test

amitsela opened the pull request.

amitsela commented:
@tomwhite should I go ahead and implement Unbound read/write (Hadoop, Text, Avro) ? or you think it's better to create an issue and wait for Dataflow implementation for TextIO Unbound ?

tomwhite commented on src/main/java/com/cloudera/dataflow/spark/streaming/StreamingTransformTranslator.java:
```diff
@@ -155,6 +156,27 @@ public void evaluate(CreateStream.QueuedValues<T> transform, EvaluationContext
+      @SuppressWarnings("unchecked")
+      @Override
+      public void evaluate(Flatten.FlattenPCollectionList<T> transform, EvaluationContext context) {
+        PCollectionList<T> pcs = ((StreamingEvaluationContext) context).getInput(transform);
```
Define a variable for `StreamingEvaluationContext` to avoid having to repeatedly cast.

tomwhite commented on src/main/java/com/cloudera/dataflow/spark/streaming/StreamingTransformTranslator.java:
```diff
@@ -155,6 +156,27 @@ public void evaluate(CreateStream.QueuedValues<T> transform, EvaluationContext
+        JavaDStream<WindowedValue<T>> first =
+            (JavaDStream<WindowedValue<T>>) ((StreamingEvaluationContext) context)
+            .getStream(pcs.get(0));
+        JavaDStream<WindowedValue<T>>[] rest = new JavaDStream[pcs.size() - 1];
```
Use a `List` here to avoid having to convert from an array later. Also it avoids dealing with index arithmetic (`i - 1`).

tomwhite commented:
A couple of minor issues, but otherwise looks good to me.

Regarding TextIO/AvroIO/HadoopIO - reusing code would be good if possible. Does the code for TextIO Unbound already exist in Dataflow?

amitsela commented:
I don't think TextIO Unbound exists in Dataflow yet. 
https://github.com/GoogleCloudPlatform/DataflowJavaSDK/issues/81 mentions that TextIO does not work in streaming mode.

amitsela commented:
@tomwhite I've fixed the repetitive casting of StreamingEvaluationContext throughout StreamingTransformTranslator and I've used a List instead of Array like you mentioned.
I've squashed to a single commit.

tomwhite commented:
Thanks @amitsela. Regarding TextIO I would wait for the SDK issue to get done first (to avoid duplicating effort and/or implementing something with slightly different semantics).
